server/app/controllers/applicant/ApplicantProgramsController.java

```diff
@@ -148,18 +149,49 @@ public CompletionStage<Result> indexWithApplicantId(
             });
   }
 
-  @Secure
+  /**
+   * When the user is not logged in or tied to a guest account, show them the list of publicly
+   * viewable programs.
+   */
+  public CompletionStage<Result> indexWithoutApplicantId(Request request, List<String> categories) {
+    // get the programs for the active version
+    // turn them into applicationPrograms
+    CompletableFuture<ApplicationPrograms> programsFuture =
+        applicantService.relevantProgramsForNoApplicant(request).toCompletableFuture();
+
+    return programsFuture.thenApplyAsync(
+        programs -> {
+          return settingsManifest.getNorthStarApplicantUi(request)
+              ? ok(northStarProgramIndexView.render(
+                      messagesApi.preferred(request),
+                      request,
+                      Optional.empty(),
+                      ApplicantPersonalInfo.ofGuestUser(),
+                      programsFuture.join(),
+                      request.flash().get(FlashKey.BANNER),
+                      Optional.empty()))
+                  .as(Http.MimeTypes.HTML)
+              : ok(
+                  programIndexView.renderWithoutApplicant(
+                      messagesApi.preferred(request),
+                      request,
+                      programs,
+                      ImmutableList.copyOf(categories)));
+        });
+  }
+
   public CompletionStage<Result> index(Request request) {
+    if (profileUtils.optionalCurrentUserProfile(request).isEmpty()) {
+      return indexWithoutApplicantId(request, ImmutableList.of());
```

Comment:
Ok cool. If I understand correctly, preventing it from creating a session when filtered means if the previous user did any filtering and then walked away, the next person will see the filtered view. But it's pretty clear in the UI when it's filtered so I don't think that matters. What do you think?

Re: removing the `categories` param from `indexWithoutApplicantId` -- I'm suggesting instead of passing down an empty list everywhere you call `indexWithoutApplicantId`, just pass in an empty list here:
```
ok(
                  programIndexView.renderWithoutApplicant(
                      messagesApi.preferred(request),
                      request,
                      programs,
                      ImmutableList.of()));
```

I don't think this would break anything categories related, but if I'm wrong or if you would rather just leave it as is feel free to resolve this!